src/lib/shell/commands/WiFi.cpp

```diff
@@ -104,13 +109,39 @@ static CHIP_ERROR WiFiModeHandler(int argc, char ** argv)
 
 static CHIP_ERROR WiFiConnectHandler(int argc, char ** argv)
 {
-    if (argc != 2)
-    {
-        return CHIP_ERROR_INVALID_ARGUMENT;
-    }
+    CHIP_ERROR error = CHIP_NO_ERROR;
+    uint8_t networkIndex;
+    MutableCharSpan debugText;
+    ByteSpan ssidSpan;
+    ByteSpan passwordSpan;
+
+    VerifyOrReturnError(GetWiFiDriver() != nullptr, CHIP_ERROR_NOT_IMPLEMENTED);
+
+    /* Command accepts running with SSID and password as parameters */
+    VerifyOrReturnError((argc == 2), CHIP_ERROR_INVALID_ARGUMENT);
+
+    ssidSpan     = ByteSpan(reinterpret_cast<const uint8_t *>(argv[0]), strlen(argv[0]));
+    passwordSpan = ByteSpan(reinterpret_cast<const uint8_t *>(argv[1]), strlen(argv[1]));
+
+    VerifyOrReturnError(IsSpanUsable(ssidSpan) && IsSpanUsable(passwordSpan), CHIP_ERROR_INVALID_ARGUMENT);
+
+    ChipLogProgress(Shell, "Adding/Updating network %s", argv[0]);
+
+    /* AddOrUpdateNetwork() checks ssid length and password length. The network info is not persistent. */
+    GetWiFiDriver()->AddOrUpdateNetwork(ssidSpan, passwordSpan, debugText, networkIndex);
```

Comment:
debugText will be written to by the callee, but we are not providing any space to write into.  Why not?  The API contract for `WiFiDriver` doesn't seem to say that passing a zero-size buffer is allowed.